test/test_tensor.h

```diff
@@ -88,16 +70,48 @@ TEST(tensor, view1) {
   EXPECT_EQ(tensor.host_at(0, 0, 0, 0), -1);
 }
 
+TEST(tensor, view2) {
+  Tensor<> tensor({3, 3, 3, 4}, 2);
+  auto t_view = tensor.subView(TensorRange(0, 2), TensorSingleIndex(0),
+                               TensorRange(0, 3), TensorAll());
+  EXPECT_EQ(t_view.shape()[0], size_t(2));
+  EXPECT_EQ(t_view.shape()[1], size_t(3));
+  EXPECT_EQ(t_view.shape()[2], size_t(4));
```

Comment:
It is xtensor behavior, and I believe it's pretty intuitive: taking the index you expect to get rid of the dimension. I believe it's possible to use `Range` to keep axis of size 1.